File: python/p034.py
"""
	
Project Euler - Problem 34 - Digit factorials
https://projecteuler.net/problem=34

"""
import timing
from math import factorial

def main():
    nums = []
    for n in range(3,50000):
        total = sum([factorial(int(s)) for s in [x for x in str(n)]])
        if total == n:
            nums.append(n)
    
    print(sum(nums))

if __name__ == '__main__':
	main()

File: js/p034.js
"use strict";
var factorial = require("./util/tools").factorial,
    integerDigits = require("./util/tools").integerDigits;

/*
 * Digit factorials
 * Problem 34
 * 145 is a curious number, as 1! + 4! + 5! = 1 + 24 + 120 = 145.
 * 
 * Find the sum of all numbers which are equal to the sum of the factorial of 
 * their digits.
 * 
 * Note: as 1! = 1 and 2! = 2 are not sums they are not included.
 * 
 * Answer: 40730
 */

function init() {
    var f = {};
    for (var i=0; i<=9; i++) 
    {
        f[i] = factorial(i);
    }

    var output = [];
    
    for (var i=3; i<100000; i++) {
        var s = 0;
        var d = integerDigits(i);
        for (var j=0; j<d.length; j++) {
            s += f[d[j]];
        }

        if (s == i) {
            output.push(i);
        }
    }
    return output.reduce((a,b) => a+b);
}

module.exports.init = init;